drivers/mpu9250/MPU9250.cpp

```diff
@@ -232,51 +125,62 @@ int MPU9250::mpu9250_init()
 	 */
 	//uint8_t samplerate_divider = 0;
 	//result = _writeReg(MPUREG_FIFO_EN, samplerate_divider);
-
 	//if (result != 0) {
 	//	DF_LOG_ERR("sample rate config failed");
 	//}
-
 	//usleep(1000);
-
 	result = _writeReg(MPUREG_CONFIG,
-			   BITS_DLPF_CFG_250HZ |
-			   BITS_CONFIG_FIFO_MODE_OVERWRITE);
+			BITS_DLPF_CFG_250HZ | BITS_CONFIG_FIFO_MODE_OVERWRITE);
 
 	if (result != 0) {
 		DF_LOG_ERR("config failed");
 	}
 
 	usleep(1000);
 
-	result = _writeReg(MPUREG_GYRO_CONFIG,
-			   BITS_FS_2000DPS |
-			   BITS_BW_LT3600HZ);
+	result = _writeReg(MPUREG_GYRO_CONFIG, BITS_FS_2000DPS | BITS_BW_LT3600HZ);
 
 	if (result != 0) {
 		DF_LOG_ERR("Gyro scale config failed");
 	}
 
 	usleep(1000);
 
-	result = _writeReg(MPUREG_ACCEL_CONFIG,
-			   BITS_ACCEL_CONFIG_16G);
+	result = _writeReg(MPUREG_ACCEL_CONFIG, BITS_ACCEL_CONFIG_16G);
 
 	if (result != 0) {
 		DF_LOG_ERR("Accel scale config failed");
 	}
 
 	usleep(1000);
 
-	result = _writeReg(MPUREG_ACCEL_CONFIG2,
-			   BITS_ACCEL_CONFIG2_BW_1130HZ);
+	result = _writeReg(MPUREG_ACCEL_CONFIG2, BITS_ACCEL_CONFIG2_BW_1130HZ);
 
 	if (result != 0) {
 		DF_LOG_ERR("Accel scale config2 failed");
 	}
 
 	usleep(1000);
 
+	// Initialize the magnetometer inside the IMU, if enabled by the caller.
+	if (_mag_enabled && _mag == nullptr) {
+		if ((_mag = new MPU9250_mag(*this, MPU9250_MAG_SAMPLE_RATE_100HZ))
```

Comment:
Since there is a `new` here, I think, there should be a `delete` in the destructor.